catalogs/catalog-lakehouse-iceberg/build.gradle.kts

```diff
@@ -31,12 +31,22 @@ val icebergVersion: String = libs.versions.iceberg.get()
 val scalaCollectionCompatVersion: String = libs.versions.scala.collection.compat.get()
 
 dependencies {
-  implementation(project(":api"))
+  implementation(project(":api")) {
+    exclude("*")
+  }
   implementation(project(":catalogs:catalog-common"))
-  implementation(project(":common"))
-  implementation(project(":core"))
-  implementation(project(":iceberg:iceberg-common"))
-  implementation(project(":server-common"))
+  implementation(project(":common")) {
+    exclude("*")
+  }
+  implementation(project(":core")) {
+    exclude("*")
+  }
+  implementation(project(":iceberg:iceberg-common")) {
```

Comment:
some jars in `:iceberg:iceberg-common` are critical, we couldn't exclude all the dependences,  we could exclude extra jars in the build file of  `:iceberg:iceberg-common`